packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -310,6 +313,23 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         }
     }
 
+    public async tombstone() {
+        this.verifyNotClosed();
+        if (this.tombstoned) {
+            return;
+        }
+
+        assert(!(await this.isRoot()), "Tombstone should not be called on root datastores!");
+        assert(!this.isLocalDataStore, "Local data should have been wiped before tombstone is called!");
+
+        // Note loaded and channel !== undefined are the same thing as you are only loaded if the channel is defined
+        assert(!this.loaded || this.channel === undefined, "Cannot tombstone a loaded datastore.");
+        assert(this.pending === undefined || this.pending.length === 0,
```

Comment:
Why can't a data store with pending op be tombstoned?

Reply:
That would mean ops are being sent to this datastore and could potentially corrupt the document right?

Reply:
A data store should not get ops after it has been tombstoned. It's not ideal but It is fine for it to get ops before that. The whole idea behing tombstone is to add a point in time after which it cannot happen so we should not assert on pending ops here.